R/allelic_series_test.R

```diff
@@ -461,44 +459,55 @@ COAST <- function(
     return_null_model = TRUE,
     weights = weights
   )
-  p_skat <- c(p_allelic_skat = skat_list$p)
+  p_skat <- c(allelic_skat = skat_list$p)
   null <- skat_list$null
 
   # Standard SKAT-O all.
   if (include_orig_skato_all) {
     orig_skato_all <- SKAT::SKAT(geno, null, method = "SKATO")
-    p_skat <- c(p_skat, p_orig_skat_all = orig_skato_all$p.value)
+    p_skat <- c(p_skat, orig_skat_all = orig_skato_all$p.value)
   }
 
   # Standard SKAT-O PTV.
   if (include_orig_skato_ptv) {
     ptv <- geno[, anno == 2, drop = FALSE]
     if (ncol(ptv) > 0) {
       orig_skato_ptv <- SKAT::SKAT(ptv, null, method = "SKATO")
-      p_skat <- c(p_skat, p_orig_skat_ptv = orig_skato_ptv$p.value)
+      p_skat <- c(p_skat, orig_skat_ptv = orig_skato_ptv$p.value)
     }
   }
 
   # Omnibus.
   n_burden <- length(p_burden)
   n_skat <- length(p_skat)
 
-  p_val = c(p_burden, p_skat)
+  pvals = c(p_burden, p_skat)
   omni_weights <- c(rep(1, n_burden), rep(n_burden / n_skat, n_skat))
 
-  p_omni <- RNOmni::OmniP(p = p_val, w = omni_weights)
+  p_omni <- RNOmni::OmniP(p = pvals, w = omni_weights)
+  pvals <- c(pvals, omni = p_omni)
 
-  # Output.
+  # Only omnibus p-value requested.
   if (return_omni_only) {
```

Comment:
n.b. any reason not to move this up above the `pvals <- c(...)` line and return one step earlier?

Reply:
Makes sense, updated as suggested.